libs/api/libbpf_program.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "api_internal.h"
+#pragma warning(push)
+#pragma warning(disable : 4200)
+#include "libbpf.h"
+#pragma warning(pop)
+#include "libbpf_internal.h"
+
+// This file implements APIs in LibBPF's bpf.h
+// and is based on code in libbpf.c.
+
+static const ebpf_program_type_t*
+_get_ebpf_program_type(enum bpf_prog_type type)
+{
+    // TODO(issue #223): read this mapping from the registry
+    switch (type) {
+    case BPF_PROG_TYPE_XDP:
+        return &EBPF_PROGRAM_TYPE_XDP;
```

Comment:
EBPF_PROGRAM_TYPE_XDP definition which is present in ebpf_api.h is temporary till Issue #223 is fixed. Once these definitions are removed, how will `bpf_prog_type` enums be mapped to the program type guids?

Reply:
Reading the mapping from the registry, as line 17 says